crates/bevy_render/src/mesh/allocator.rs

```diff
@@ -955,12 +956,8 @@ impl ElementLayout {
     /// Creates an [`ElementLayout`] for mesh data of the given class (vertex or
     /// index) with the given byte size.
     fn new(class: ElementClass, size: u64) -> ElementLayout {
-        const {
-            assert!(4 == COPY_BUFFER_ALIGNMENT);
-        }
-        // this is equivalent to `4 / gcd(4,size)` but lets us not implement gcd.
-        // ping @atlv if above assert ever fails (likely never)
-        let elements_per_slot = [1, 4, 2, 4][size as usize & 3];
+        let elements_per_slot =
+            n_over_gcd_by_table::<{ COPY_BUFFER_ALIGNMENT as usize }>(size as usize) as u32;
```

Comment:
Now that I'm looking at it, is there a possibility of a bug caused by usize being smaller than u64?
Can the input values reasonably be more than u32? On a 32-bit platform?

Reply:
I've refactored the `gcd` implementations to work on `u64` explicitly, which moves the casts there. By casting from `usize` to `u64`, we avoid size issues on all platforms less than 128-bit.